src/Ledgers.jl

```diff
@@ -31,7 +32,8 @@ mutable struct Account{P<:Position}
 end
 Account(balance::Position) = Account(AccountId(uuid4()), balance)
 
-struct AccountInfo{AT<:AccountType,A<:Account}
+abstract type AccountNode end
+struct AccountInfo{AT<:AccountType,A<:Account} <: AccountNode
```

Comment:
Just remove the AccountType here, it's not used.

Reply:
I don't think you need A<:Account either

Reply:
I agree this needs to be revisited, but let's look at it in a separate PR.